In [1]:
from keras.preprocessing.image import load_img

In [3]:
# load the image
img = load_img('bondi_beach.jpg')
# report details about the image
print(type(img))
print(img.format)
print(img.mode)
print(img.size)
# show the image in windows default photo application 
img.show() 

<class 'PIL.JpegImagePlugin.JpegImageFile'>
JPEG
RGB
(640, 427)


**We can confirm that the image was loaded as a PIL image in JPEG format with RGB channels and the size of 640 by 427 pixels**

# Convert Image to array

In [4]:
from keras.preprocessing.image import load_img
from keras.preprocessing.image import img_to_array
from keras.preprocessing.image import array_to_img

In [5]:
img = load_img('bondi_beach.jpg')

In [6]:
img_array = img_to_array(img)

In [8]:
print(type(img_array))
print(img_array.shape)

<class 'numpy.ndarray'>
(427, 640, 3)


# Convert Array to Image

In [9]:
img_pil = array_to_img(img_array)

In [11]:
print(type(img_pil))
img_pil.show() 

<class 'PIL.Image.Image'>


# Save Image

In [12]:
from keras.preprocessing.image import load_img
from keras.preprocessing.image import save_img
from keras.preprocessing.image import img_to_array

In [16]:
# Import image as Greyscale
img = load_img('bondi_beach.jpg', grayscale=True)

In [17]:
img_array = img_to_array(img)

In [18]:
print(type(img_array))
print(img_array.shape)

<class 'numpy.ndarray'>
(427, 640, 1)


In [19]:
save_img('bondi_beach_grayscale.jpg', img_array)